Objects/dictobject.c

```diff
@@ -1687,46 +1832,68 @@ PyDict_GetItem(PyObject *op, PyObject *key)
 
 Py_ssize_t
 _PyDict_GetItemHint(PyDictObject *mp, PyObject *key,
-                    Py_ssize_t hint, PyObject **value)
+                    Py_ssize_t hint, PyObject **value_addr)
```

Comment:
The first commit in the "stack" of commits is a fix of this one, it has several things that yes, aren't related to lazy imports, but ended up being needed for consistency of the code. For example, there can't be split dictionaries with non-unicode keys, and they're assuming there could be. Just wanted to be consistent with the other parts of the codebase and the use of Lazy Imports (to avoid other problems)